bika/lims/browser/stickers.py

```diff
@@ -27,12 +27,23 @@ class Sticker(BrowserView):
 
     def __call__(self):
         self.rendered_items = []
-        bc = getToolByName(self.context, 'bika_catalog')
+        # Trying to find a catalog for given content type.
+        content_type = self.request.get('ctype', '')
+        catalog = None
+        if content_type:
+            archetype_tool = getToolByName(self.context, 'archetype_tool')
+            catalogs_list = archetype_tool.catalog_map.get(content_type, [])
+            if len(catalogs_list) > 0:
+                catalog_id = catalogs_list[0]
+                catalog = getToolByName(self.context, catalog_id)
+        # Just in case, if no catalog found, use bika_catalog
+        if not catalog:
+            catalog = getToolByName(self.context, 'bika_catalog')
```

Comment:
Better to add a function in `bika.lims.catalog` to allow us to reuse the code for obtaining the catalog to which a given content_type lives by default. Note there is already a function called `getCatalog` in `bika.lims.catalog`: https://github.com/naralabs/bika.lims/blob/wip/bika/lims/catalog.py#L170 , but the param in required is an instance rather to a string representation of the content_type. Maybe you could add a function like `getCatalogByPortalType` and reuse it inside the existing function.